### PR TITLE
[RDY] Backport Trophy Fix (0.65)

### LINKED ISSUE #2004: [Bug] AllCuresTrophy crashes existing saves
#### The issue
While playing the first hospital, having reached all my goals (10 cured patients and a rating above 200), time freezes on the 31st of December, right before I get the letter to move on to the second hospital. I can still build and place things, but the people are frozen, time doesn't progress and I don't get my letter to go to the next level. Moreover, if I quit out onto the main menu, my cursor freezes in place. And the only way to visibly move it is to click somewhere else.

#### Steps to Reproduce
1. Load the save file attached.

2. See that it's frozen in time.

3. Quit out onto the main menu and observe that the cursor is stuck.

#### Save Game
[CorsixTH save.zip](https://github.com/CorsixTH/CorsixTH/files/6796561/CorsixTH.save.zip)

#### Expected Behavior
I expected the letter to move on to the next hospital, or at least that the game would continue playing.

#### System Information
**CorsixTH Version:** 0.65

**Operating System:** Windows 10

**Theme Hospital Version:** GOG.com

#### Gamelog.txt
Gamelog.txt comes within the .zip file included.

### Additional Info
I don't know if this is related, but my (local) account name on my laptop has a Romanian ș in it, and the Users folder has the Turkish ş in it. This has caused problems for me in the past (I still can't play OpenLoco on this account, and I've already run into countless issues with CorsixTH). It might also be my laptop that's bungling up something. I never had the same problems with CorsixTH on my older laptop (or any program not working because of my username), despite it having the same account name. Thought you might wanna know.

### PR DESCRIPTION
**Pull request by lewri**
*Fixes release branch of #2004*

**Describe what the proposed change does**
- Apply the correct config params for 0.65 release for the All Cures Trophy.

### CODE REVIEW

**Comment by lewri** [comment]
I'll likely make a Wiki page for a standard hotfix format if this gets approved.

File: CorsixTH/Lua/map.lua [hunk 1]
```diff
@@ -802,4 +821,8 @@ function Map:afterLoad(old, new)
       self:setCellFlags(58, 72, {passable = false})
     end
   end
+  -- 0.65 has a hotfix, make sure we patch any affected games
+  if not self.hotfix1 then
```
Comment (TheCycoONE) [inline CorsixTH/Lua/map.lua:825]: Interesting - I approve

**Comment by TheCycoONE** [comment]
LGTM, add 0.65.1 to the changelog too if you don't mind.

**Comment by lewri** [comment]
Would 0.65.1 break ``checkForUpdates``? (We could likely use check for updates to spread the hotfix, but it might have to be 0.651, unless I don't understand how the version compare works)

**Comment by TheCycoONE** [comment]
No checkForUpdates uses a 3 part vision - which is good because there's no way I'd use 0.651 - it's number.number not a decimal.

**Comment by TheCycoONE** [comment]
That said you will have to change the version name in getVersion as well.

**Comment by lewri** [comment]
Thanks! I'll have a stab at it and let me know if I did it wrong

**Comment by lewri** [comment]
Wiki page made, feel free to edit as needed https://github.com/CorsixTH/CorsixTH/wiki/Hotfixes

File: CorsixTH/com.corsixth.corsixth.metainfo.xml [hunk 2]
```diff
@@ -26,6 +26,15 @@
   </provides>
   <launchable type="desktop-id">com.corsixth.corsixth.desktop</launchable>
   <releases>
+    <release version="0.65.1" date="2021-07-13" type="stable">
+      <description>
+        <p>This is a hotfix release of 0.65 and replaces the June 2021 release</p>
+        <ul>
+          <li>Fixes a bug where the game would crash if you got the All Cured trophy at the
```
Comment (TheCycoONE) [inline CorsixTH/com.corsixth.corsixth.metainfo.xml:33]: Important context is that this is only when loading a save from 0.64

Comment (lewri) [reply CorsixTH/com.corsixth.corsixth.metainfo.xml:33]: Its not 0.64 specific. But point noted I need to say for existing saves

File: changelog.txt [hunk 3]
```diff
@@ -1,3 +1,10 @@
+-------------------------------------------------------------------------------
+Version 0.65.1 - released July 2021
+-------------------------------------------------------------------------------
+# This is a hotfix release of 0.65 and replaces the June 2021 release
+* Fixes a bug where the game would crash if you got the All Cured trophy at the
+  end of year awards
```
Comment (lewri) [inline changelog.txt:6]:
```suggestion
  end of year awards, when loading a save from 0.64 or earlier
```

File: CorsixTH/com.corsixth.corsixth.metainfo.xml [hunk 4]
```diff
@@ -26,6 +26,15 @@
...
+            end of year awards</li>
```
Comment (lewri) [inline CorsixTH/com.corsixth.corsixth.metainfo.xml:34]:
```suggestion
            end of year awards, when loading a save from 0.64 or earlier</li>
```

File: CorsixTH/com.corsixth.corsixth.metainfo.xml [hunk 5]
```diff
@@ -26,6 +26,15 @@
   </provides>
   <launchable type="desktop-id">com.corsixth.corsixth.desktop</launchable>
   <releases>
+    <release version="0.65.1" date="2021-07-13" type="stable">
```
Comment (lewri) [inline CorsixTH/com.corsixth.corsixth.metainfo.xml:29]:
```suggestion
    <release version="0.65.1" date="2021-07-17" type="stable">
```

**Pull request merged**